Source/utils/static_bit_vector.hpp

```diff
@@ -0,0 +1,92 @@
+#include <algorithm>
+#include <climits>
+#include <cstddef>
+
+#include "utils/attributes.h"
+
+namespace devilution {
+
+/**
+ * @brief A stack-allocated bit-vector with a fixed capacity and a fixed size < capacity.
+ *
+ * @tparam N capacity.
+ */
+template <size_t N, typename Word = unsigned long> // NOLINT(google-runtime-int)
+class StaticBitVector {
+public:
+	explicit StaticBitVector(size_t size)
+	    : size_(size)
+	    , num_words_(size / BitsPerWord + ((size % BitsPerWord) == 0 ? 0 : 1))
+	{
+	}
+
+	[[nodiscard]] bool test(size_t pos) const { return (word(pos) & maskBit(pos)) != 0; }

```

Comment:
I guess this should solve the signed compare:
```suggestion
	[[nodiscard]] bool test(size_t pos) const { return (word(pos) & maskBit(pos)) != 0u; }
```

Reply:
Fixed (it was because `maskBit` argument was `unsigned` rather than `size_t`)